web/views/log_entry.py

```diff
@@ -0,0 +1,129 @@
+        query_filters.update({"timestamp__date__lte": end_date})
+
+        if "action" in filters.keys():
+            query_filters.update({"action__exact": filters.get("action")})
+
+        if "entity_name" in filters.keys() and "entity_id" in filters.keys():
+            entity_class = get_object_or_404(ContentType, model=filters.get("entity_name")).model_class()
+            entity_object = get_object_or_404(entity_class, pk=filters.get("entity_id"))
+            self.object_list = [
+                {"action": log.Action.choices[log.action], "log": log}
+                for log in entity_object.history.filter(**query_filters).all()
+            ]
+
+        else:
+            if "entity_name" in filters.keys():
+                query_filters.update({"content_type__model": filters.get("entity_name")})
+                if "parent_entity_name" in filters.keys() and "parent_entity_id" in filters.keys():
+                    entity_class = get_object_or_404(ContentType, model=filters.get("entity_name")).model_class()
+                    entity_ids_list = [
+                        o.pk for o in entity_class.objects.filter(**{filters.get("parent_entity_name"): filters.get("parent_entity_id")})
+                    ]
+                    query_filters.update({"object_id__in": entity_ids_list})
+
+                if "entity_attr" in filters.keys():
+                    field = filters.get("entity_attr")
+                    query_filters.update({"changes__regex": rf'"{field}": \['})
+
+            if "user" in filters.keys():
+                query_filters.update({"actor__exact": filters.get("user")})
+
+            print(f"Filters: {query_filters}")
+            self.object_list = [{"action": log.Action.choices[log.action], "log": log} for log in LogEntry.objects.filter(**query_filters).all()]
+
+        context = super().get_context_data(**kwargs)
+
+        models_list_fk = LogEntry.objects.values("content_type").distinct().all()
+        models_list_names = [ct["model"] for ct in ContentType.objects.values("model").filter(pk__in=models_list_fk).all()]
+        users_list_fk = LogEntry.objects.values("actor").distinct().all()
+        users_list_names = User.objects.values("pk", "full_name").filter(pk__in=users_list_fk).all()
+        context["models_list"] = models_list_names
+        context["users_list"] = users_list_names
+        context["start_date"] = start_date.strftime(self.DATE_FORMAT)
+        context["end_date"] = end_date.strftime(self.DATE_FORMAT)
+        context["log_actions"] = LogEntry.Action.choices
+        context["model_fields"] = self.get_list_of_model_fields()
+        return context
+
+    def check_permissions(self, request):
+        if request.user.is_superuser or request.user.is_part_of(constants.Groups.DATA_STEWARD.value) or request.user.is_part_of(constants.Groups.AUDITOR.value):
+            return None
+        else:
+            if "entity_name" in request.GET.keys() and "entity_id" in request.GET.keys():
```

Comment:
if .GET is a dict, then `in request.GET` should already check if the key exists, no need for the `.keys()`

Reply:
Forgot about that.
I changed, also as I test for the presence of an entry in the keys I removed the .get() in following lines.